src/test/interface/file_test.cpp

```diff
@@ -213,3 +213,16 @@ TEST(CommandHelper, check_filename_config_bad3) {
 
   EXPECT_THROW(check_file_config(parser), std::invalid_argument);
 }
+
+TEST(CommandHelper, check_same_file_test) {
```

Comment:
Thanks for the test. Adding at least one `EXPECT_TRUE` seems like a good idea to show the cases it does actually handle. This test would pass with a function that always returned false 😉